controller/controller.go

```diff
@@ -260,6 +300,68 @@ func (c controller) GetStatus(req Request) ([]fleet.UnitStatus, error) {
 	return status, maskAny(err)
 }
 
+func (c controller) WaitForStatus(req Request, desired Status, closer <-chan struct{}) error {
+	fail := make(chan error)
+	done := make(chan struct{})
+
+	go func() {
+		// c describes the count of how often the desired aggregated status was
+		// seen.
+		count := 0
+
+	L1:
+		for {
+			unitStatusList, err := c.groupStatus(req)
+			if IsUnitNotFound(err) && desired == StatusNotFound {
+				goto C1
+			} else if err != nil {
+				fail <- maskAny(err)
+				return
+			}
+
+			for _, us := range unitStatusList {
+				for _, ms := range us.Machine {
+					aggregated, err := AggregateStatus(us.Current, us.Desired, ms.SystemdActive, ms.SystemdSub)
+					if err != nil {
+						fail <- maskAny(err)
+						return
+					}
+
+					if aggregated != desired {
+						// Whenever the aggregated status does not match the desired
+						// status, we reset the counter.
+						count = 0
+						time.Sleep(c.WaitSleep)
+						continue L1
+					}
+				}
+			}
+
+		C1:
+			count++
+			if count == 3 {
```

Comment:
Note that 3 times is a magic number out of my head by plain guessing. Don't know what is more sufficient.